configure.sh

```diff
@@ -1,171 +1,21 @@
-  rm -f "$BASE_PATH/$PRIVATE_GOOGLE_SERVICES"
-}
-
-# Clone the private repo and copy all of its files (except README.md) into the main repo
-setup_private() {
-  echo "Copying private configuration files from the repo '$PRIVATE_REPO', branch '$PRIVATE_BRANCH'"
-  set -x
-  rm -rf "$TMP_REPO_DIR"
-  git clone --branch "$PRIVATE_BRANCH" --depth 1 "$PRIVATE_REPO" "$TMP_REPO_DIR"
-  echo "$PRIVATE_REPO" > "$SAVED_PRIVATE_REPO_FILE"
-  echo "$PRIVATE_BRANCH" > "$SAVED_PRIVATE_BRANCH_FILE"
-  echo "Saved private repository url '$PRIVATE_REPO' to '$SAVED_PRIVATE_REPO_FILE'"
-  echo "Saved private branch '$PRIVATE_BRANCH' to '$SAVED_PRIVATE_BRANCH_FILE'"
-  cd $TMP_REPO_DIR
-  rm -rf "$TMP_REPO_DIR/.git" "$TMP_REPO_DIR/README.md"
-  cp -Rv "$TMP_REPO_DIR"/* "$BASE_PATH"
-  rm -rf "$TMP_REPO_DIR"
-  # Remove old android secrets during the transition period to the new project structure
-  echo "Removing keys from old locations"
-  rm -f android/release.keystore \
-        android/secure.properties \
-        android/libnotify.properties \
-        android/google-services.json \
-        android/google-play.json \
-        android/firebase-app-distribution.json \
-        android/firebase-test-lab.json \
-        android/huawei-appgallery.json \
-        android/res/xml/network_security_config.xml
-  set +x
-  echo "Private files have been updated."
-}
-
-if [ "${1-}" = "-h" -o "${1-}" = "--help" ]; then
-  usage
-  exit 1
-fi
-
-ARGS_PRIVATE_REPO=${1-}
-ARGS_PRIVATE_BRANCH=${2-}
-
-if [ -n "$ARGS_PRIVATE_REPO" ]; then
-  PRIVATE_REPO=$ARGS_PRIVATE_REPO
-  if [ -n "$ARGS_PRIVATE_BRANCH" ]; then
-    PRIVATE_BRANCH=$ARGS_PRIVATE_BRANCH
-  else
-    PRIVATE_BRANCH=master
-  fi
-else
-  read -t 1 READ_PRIVATE_REPO READ_PRIVATE_BRANCH || true
-  if [ -n "${READ_PRIVATE_REPO-}" ]; then
-    PRIVATE_REPO=$READ_PRIVATE_REPO
-    if [ -n "${READ_PRIVATE_BRANCH-}" ]; then
-      PRIVATE_BRANCH=$READ_PRIVATE_BRANCH
-    else
-      PRIVATE_BRANCH=master
-    fi
-  elif [ -f "$SAVED_PRIVATE_REPO_FILE" ]; then
-    PRIVATE_REPO=`cat "$SAVED_PRIVATE_REPO_FILE"`
-    echo "Using stored private repository URL: $PRIVATE_REPO"
-    if [ -f "$SAVED_PRIVATE_BRANCH_FILE" ]; then
-      PRIVATE_BRANCH=`cat "$SAVED_PRIVATE_BRANCH_FILE"`
-      echo "Using stored private branch: $PRIVATE_BRANCH"
-    else
-      PRIVATE_BRANCH=master
-    fi
-  else
-    PRIVATE_REPO=""
-  fi
-fi
-
-if [ -n "$PRIVATE_REPO" ]; then
-  setup_private
-else
-  setup_opensource
-fi
-
-# TODO: Remove these lines when XCode project is finally generated by CMake.
-if [ ! -d "$BASE_PATH/3party/boost/" ]; then
-  echo "You need to have Boost submodule present to run bootstrap.sh"
-  echo "Try 'git submodule update --init --recursive'"
-  exit 1
-fi
-if [ ! -d "$BASE_PATH/3party/boost/tools" ]; then
-  echo "Boost's submodule 'tools' is missing, it is required for bootstrap"
-  echo "Try 'git submodule update --init --recursive'"
-  exit 1
+if [ ! -f private.h ]; then
+  cp -p private_default.h private.h
```

Comment:
is it needed at all?

e.g. we don't keep `gradle.properties` in .gitignore, though some people might change it to suit their needs